libs/community/langchain_community/chat_models/notdiamond.py

```diff
@@ -0,0 +1,416 @@
+    max_retries: int = 6
+    model_kwargs: Dict[str, Any] = Field(default_factory=dict)
+    """Holds any model parameters valid for `create` call not explicitly specified."""
+
+    @pre_init
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate api key and python package exists."""
+        try:
+            from notdiamond import NotDiamond
+        except ImportError:
+            raise ChatNotDiamondException(
+                "Could not import notdiamond python package. "
+                "Please install it with `pip install notdiamond[create]`"
+            )
+        
+        values["notdiamond_api_key"] = get_from_dict_or_env(
+            values, "notdiamond_api_key", "NOTDIAMOND_API_KEY", default=""
+        )
+        values["openai_api_key"] = get_from_dict_or_env(
+            values, "openai_api_key", "OPENAI_API_KEY", default=""
+        )
+        values["anthropic_api_key"] = get_from_dict_or_env(
+            values, "anthropic_api_key", "ANTHROPIC_API_KEY", default=""
+        )
+        values["google_api_key"] = get_from_dict_or_env(
+            values, "google_api_key", "GOOGLE_API_KEY", default=""
+        )
+        values["mistral_api_key"] = get_from_dict_or_env(
+            values, "mistral_api_key", "MISTRAL_API_KEY", default=""
+        )
+        values["replicate_api_key"] = get_from_dict_or_env(
+            values, "replicate_api_key", "REPLICATE_API_KEY", default=""
+        )
+        values["together_api_key"] = get_from_dict_or_env(
+            values, "together_api_key", "TOGETHER_API_KEY", default=""
+        )
+        values["perplexity_api_key"] = get_from_dict_or_env(
+            values, "perplexity_api_key", "PPLX_API_KEY", default=""
+        )
+        values["cohere_api_key"] = get_from_dict_or_env(
+            values, "cohere_api_key", "COHERE_API_KEY", default=""
+        )
+        notdiamond_client = NotDiamond(llm_configs=values["llm_configs"], api_key=values["notdiamond_api_key"])
+        values["client"] = notdiamond_client
+        return values
+    
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        """Get the default parameters for calling Not Diamond's API."""
+        return {
+            "model": self.llm_configs,
+            "default": self.default,
+            "max_model_depth": self.max_model_depth,
+            "latency_tracking": self.latency_tracking,
+            "hash_content": self.hash_content,
+            "tradeoff": self.tradeoff,
+            "preference_id": self.preference_id,
+            # "tools": self.tools,
+            **self.model_kwargs,
+        }
+
+    @property
+    def _identifying_params(self) -> Dict[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            "llm_configs": self.llm_configs,
+            "tradeoff": self.tradeoff,
+            "tools": self.tools,
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        return "notdiamond-chat"
```

Comment:
Semantically this is too similar to our chat offering - let's just remove the suffix

```suggestion
        return "notdiamond"
```

Reply:
Might want to check if there are requirements for integrating into the chat category of langchain